client/src/document.js

```diff
@@ -100,69 +100,93 @@ function RenderSideBar({ doc }) {
-                <Link to={node.uri}>{node.title}</Link>
-              </li>
-            );
-          }
-        })}
-      </ul>
-    </div>
-  );
-}
-
-function SidebarLeaflets({ node }) {
+function SidebarLeaf({ parent }) {
+  const titleNode = <h3>{parent.title}</h3>;
   return (
     <Location>
       {({ location }) => {
-        let hasActiveChild = false;
-        const listItems = node.content.map(childNode => {
-          const isActive = childNode.uri === location.pathname;
-          if (isActive && !hasActiveChild) {
-            hasActiveChild = true;
+        // walk through the tree, annotating its nodes:
+        // * `isActive` if a node contains the link to the current page
+        // * `open` if a node is on the path to the `isActive` node
+        function setOpenNodes(node, pathname) {
+          if (node.uri === pathname) {
+            node.open = true;
+            node.isActive = true;
+            return true;
+          }
+          if (node.content) {
+            for (const child of node.content) {
+              if (setOpenNodes(child, pathname)) {
+                node.open = true;
+                return true;
+              }
+            }
           }
-          return (
-            <li key={childNode.uri} className={isActive ? "active" : undefined}>
-              <Link to={childNode.uri}>{childNode.title}</Link>
-            </li>
-          );
-        });
+          return false;
+        }
+        setOpenNodes(parent, location.pathname);
 
         return (
-          <details open={!!hasActiveChild}>
-            <summary>{node.title}</summary>
-            <ol>{listItems}</ol>
-          </details>
+          <div>
+            {titleNode}
+            <ul>
+              {parent.content.map(node => {
+                if (node.content) {
+                  return (
+                    <li key={node.title}>
+                      <SidebarLeaflets node={node} />
+                    </li>
+                  );
+                } else {
+                  return (
+                    <li key={node.uri}>
+                      <Link to={node.uri}>{node.title}</Link>
+                    </li>
+                  );
+                }
+              })}
+            </ul>
+          </div>
         );
       }}
     </Location>
   );
 }
 
+function SidebarLeaflets({ node }) {
+  const listItems = node.content.map(childNode => {
+    if (childNode.content) {
+      return (
+        <li key={childNode.title}>
+          <SidebarLeaflets node={childNode} />
+        </li>
+      );
+    } else {
+      return (
+        <li
+          key={childNode.uri}
+          className={childNode.isActive ? "active" : undefined}
+        >
+          <Link to={childNode.uri}>{childNode.title}</Link>
+        </li>
+      );
+    }
+  });
+
+  return (
+    <details open={node.open}>
+      <summary>
+        {node.uri ? <a href={node.uri}> {node.title}</a> : node.title}
+      </summary>
+      <ol>{listItems}</ol>
```

Comment:
I think it's a bit of an anti-pattern in JSX to create JSX-node variables instead of inlining them. I'd much rather this was:

```jsx
      <ol>
        {node.content.map(childNode => {
          if (childNode.content) {
            return (
              <li key={childNode.title}>
                <SidebarLeaflets node={childNode} />
              </li>
            );
          } else {
            return (
              <li
                key={childNode.uri}
                className={childNode.isActive && "active"}
              >
                <Link to={childNode.uri}>{childNode.title}</Link>
              </li>
            );
          }
        })}
      </ol>
```
One of the motivations is that it immediately tells the code-reader that there's no reason why we're looping through it first. Otherwise, I might ask myself "Why did this need to be done first? Is it keeping a counter? What's the hack?"
The other reason it's better to not define jsx-node variables is the location of them. Now you have space between two different concerns and it's can cause the code-reader to have to jump around more than necessary:
```javascript
const listItems = { CONCERN 1 }
return <div>
    { CONCERN 2 }
    { CONCERN 3 }
    { CONCERN 4 }
    { CONCERN 5 }
    { CONCERN 1 }
</div>
```